h2o-core/src/main/java/water/util/MRUtils.java

```diff
@@ -116,14 +116,20 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
     @Override public void map(Chunk ys) {
       _ys = new double[_nclass];
       for( int i=0; i<ys._len; i++ )
-        if (!ys.isNA(i))
-          _ys[(int) ys.at8(i)]++;
+        if (!ys.isNA(i)) {
+          // PUBDEV-6697 quasibinomial GBM can have different domains than {0, 1}, for example {0, 2} or {0, 42}
+          int index = ys.at8(i) >= _ys.length ? 1 : (int) ys.at8(i);
```

Comment:
If the domains were different than {0,1} for example {0,2}, it caused OutOfArrayIndexException here. It tried to reach `_ys[2]`, but the length of `_ys` is always 2 for quasibinomial.